src/DIRAC/Core/Security/IAMService.py

```diff
@@ -87,9 +87,21 @@ def convert_iam_to_voms(iam_output):
             cert_dict = {}
             dn = convert_dn(cert["subjectDn"])
             ca = convert_dn(cert["issuerDn"])
-
             cert_dict["CA"] = ca
-            cert_dict["nickname"] = iam_output["userName"]
+
+            # The nickname is available in the list of attributes
+            # (if configured so)
+            # in the form {'name': 'nickname', 'value': 'chaen'}
+            # otherwise, we take the userName
+            try:
+                nickname = [
+                    attr["value"]
+                    for attr in iam_output["urn:indigo-dc:scim:schemas:IndigoUser"]["attributes"]
+                    if attr["name"] == "nickname"
+                ][0]
+            except (KeyError, IndexError):
+                nickname = iam_output["userName"]
+            cert_dict["nickname"] = nickname
```

Comment:
Somehow I find it more clear if it is:

```suggestion
            try:
                cert_dict["nickname"] = [
                    attr["value"]
                    for attr in iam_output["urn:indigo-dc:scim:schemas:IndigoUser"]["attributes"]
                    if attr["name"] == "nickname"
                ][0]
            except (KeyError, IndexError):
                cert_dict["nickname"] = iam_output["userName"]
```

Reply:
if you like. Done